contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -1191,16 +1176,16 @@ contract LoanTokenLogicStandard is LoanTokenLogicStorage {
 	 * */
 	function _verifyTransfers(
 		address collateralTokenAddress,
-		address[4] memory sentAddresses,
-		uint256[5] memory sentAmounts,
+		MarginTradeStructHelpers.SentAddresses memory sentAddresses,
+		MarginTradeStructHelpers.SentAmounts memory sentAmounts,
 		uint256 withdrawalAmount
 	) internal returns (uint256 msgValue) {
 		address _wrbtcToken = wrbtcTokenAddress;
 		address _loanTokenAddress = loanTokenAddress;
-		address receiver = sentAddresses[2];
-		uint256 newPrincipal = sentAmounts[1];
-		uint256 loanTokenSent = sentAmounts[3];
-		uint256 collateralTokenSent = sentAmounts[4];
+		address receiver = sentAddresses.receiver;
+		uint256 newPrincipal = sentAmounts.newPrincipal;
+		uint256 loanTokenSent = sentAmounts.loanTokenSent;
+		uint256 collateralTokenSent = sentAmounts.collateralTokenSent;
```

Comment:
no need to initialize extra variables - the struct vars can be used directly

Reply:
Agree for the receiver.
But, for newPrincipal, lanTokenSent, collateralTokenSent is being used multiple in the rest of function execution, using struct var directly will require extra gas.

I will remove the receiver declaration, but will keep the rest as it is

Reply:
it is more clear which data is used when referring to the struct vars.  
what is advantage of using yet another local vars instead of using the struct var directly? why do you want to keep them?

Reply:
As i mentioned above, using local vars here is saving some gas